cpp/AdvancedColorImages/AdvancedColorImages/TileDrawingManager.h

```diff
@@ -0,0 +1,66 @@
+//*********************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE SOFTWARE IS PROVIDED �AS IS�, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, 
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. 
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, 
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, 
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH 
+// THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+//*********************************************************
+
+#pragma once
+
+#include "DirectXTileRenderer.h"
+
+using namespace std;
+using namespace winrt;
+using namespace Windows::Foundation::Numerics;
+using namespace Windows::Foundation;
+
+class TileDrawingManager
+{
+public:
+	TileDrawingManager();
+	~TileDrawingManager();
+	void UpdateVisibleRegion(float3 currentPosition);
+	void UpdateViewportSize(Size newSize);
+	void SetRenderer(DirectXTileRenderer* renderer);
+	DirectXTileRenderer* GetRenderer();
+
+	const static int TILESIZE = 100;
```

Comment:
>TILESIZE [](start = 18, length = 8)

Surprised to still see a TILESIZE in this version of the app, since we don't actually need to break down the rendering into tiles... the only tiling we need is the constrainedUpdate stuff that happens to avoid max texture size issues. E.g. would things work just fine if the TILESIZE were == 1?

Reply:
Well, I wanted to render(BeginDraw & EndDraw calls) content incrementally in blocks that are of certain TileSize and not 1 pixel wide at a time. Tilesize of 100 would mean less BeginDraw/EndDraw calls if the user scrolls or zooms really fast.